pygtrie.py

```diff
@@ -1220,7 +1220,14 @@ def __init__(self, *args, **kwargs):
                 named argument is not specified on the function's prototype
                 because of Python's limitations.
         """
-        self._separator = kwargs.pop('separator', '/')
+        separator = kwargs.pop('separator', '/')
+        if not isinstance(separator, str):
```

Comment:
If you want to check the type, one approach is to use the following at the beginning of the file:
```
try:
    _basestring = basestring
except NameError:
    _basestring = str
```
and then you can use `isinstance(…, _basestring)`. There already is some Python 2/3 compatibility code there.

Another option is to use solution pointed out by @vmarkovtsev: https://github.com/JioCloud/python-six/blob/master/six.py#L41. I don’t really like explicitly checking version thus I’m leaning towards code sample I’ve included above.